client/driver/mock_driver.go

```diff
@@ -1,3 +1,5 @@
+// +build nomad_test
+
```

Comment:
add an init function that adds itself to the map

```
func init() {
  BuiltinDrivers["mock_driver"] = NewMockDriver
}
```
https://golang.org/doc/effective_go.html#init